SparkPost/2web2ui
# EO-577 Create engagement rate by cohort page

Note: Give your PR a short title. Something like "_TICKET NUMBER: feature description_" is good.

### What Changed
- Renamed `/signals/engagement-recency` to `/signals/engagement/cohorts`
  - Three new routes gated with the UI option `feature_signals_v2`:
    - `/signals/engagement/engagement-rate`
    - `/signals/engagement/complaints`
    - `/signals/engagement/unsubscribes`
- New `/signals/engagement/engagement-rate` page
- New fake endpoint (API endpoint does not exist)
- Abstracted out date selection into a higher order component (hooks pls)


### How To Test
- Point your local openresty upstreams to staging
- Verify UI does not change without the `feature_signals_v2` UI option
  - Visit `/signals/engagement/cohorts`
  - Verify the Tabs do not render
  - Verify the the gated routes 404
- Verify the new page
  - Log into `appteam` or add the `feature_signals_v2` UI option to your account
  - Verify the Tabs render on `/signals/engagement/cohorts`
  - Click on Engagement Rate tab

### To Do

### Maybe To Do
- [ ] Add recommended actions
- [ ] Hook up to real API
- [x] Abstract out selected date logic, and maybe some selector reshaping to reduce copy pasted code
